workshop/Lab3/README.md

```diff
@@ -0,0 +1,291 @@
+	        image: ibmcom/guestbook:v1
+	        ports:
+	        - name: http-server
+	          containerPort: 3000
+```
+The above configuration file create a deployment object named 'guestbook' with a pod containing a single container running the image 'ibmcom/guestbook:v1'.  Also the configuration specifies replicas set to 3 and kubernetes tries to make sure that at least three active pods running at any time.
+	
+- Create guestbook deployment 
+
+   ``` $kubectl create -f guestbook-deployment.yaml```
+
+- List the pod with label app=guestbook 
+
+   ``` $kubectl get pods -l app=guestbook```
+   
+You can simply change the deployment, alter replicas using the following command. Kubernetes then sees the desired the you want and tries to reconcile the actual current state to the desired state. 
+
+When you change the number of replicas in the configuration, Kubernetes will try to add, or remove, pods from the system to match your request. To can make these modifications by using the following command:
+
+```	
+	 $kubectl edit deployment guestbook 
+```
+
+You can also edit the deployment file locally to make changes. You should use the following command to make the change effective when you edit the deployment locally.
+
+   ``` $kubectl apply -f guestbook-deployment.yaml```
+
+We now define a service object to expose the deployment to external clients.
+```
+	apiVersion: v1
+	kind: Service
+	metadata:
+	  name: guestbook
+	  labels:
+	    app: guestbook
+	spec:
+	  ports:
+	  - port: 3000
+	    targetPort: http-server
+	  selector:
+	    app: guestbook
+	  type: LoadBalancer
+``` 
+The above configuration creates a Service resource named guestbook. A Service can be used
+to create a network path for incoming traffic to your running application.
+In this case, we are setting up a route from port 3000 on the cluster to the
+"http-server" port on our app, which is port 3000 per the Deployment container spec.
+Note that we create a "LoadBalancer" type of Service which will route all incoming
+traffic to each of the instances of our application - e.g. in a round-robin algorithm.
+
+- Let us now create guestbook service.
+		``` $kubectl create -f guestbook-service.yaml```
+		
+- Test guestbook app using a browser of your choice using the url '\<your-cluster-ip\>:\<node-port\>'
+
+
+# 2. Connect to a back-end service.
+
+If you look at the guestbook source code you'll notice that it is written to support
+a variety of data stores. By default it will keep the log of guestbook entries in memory.
+That's ok for testing purposes, but as you get into a more
+"real" environment where you scale your application that model will not work because
+based on which instance the user is routed to they'll see very different results.
+To solve this we need to have all instances of our app share the same data store - 
+in this case we're going to use a redis database that we deploy to our cluster.
+
+Our first application is a simple 3-tier application
```

Comment:
This sentence seems out of place - should we delete it?

Reply:
yes